{"expand":"renderedFields,names,schema,transitions,operations,editmeta,changelog","id":"12392702","self":"https://issues.apache.org/jira/rest/api/latest/issue/12392702","key":"DERBY-3582","fields":{"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/images/icons/issuetypes/bug.png","name":"Bug","subtask":false},"timespent":null,"project":{"self":"https://issues.apache.org/jira/rest/api/2/project/10594","id":"10594","key":"DERBY","name":"Derby","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/projectavatar?pid=10594&avatarId=10122","24x24":"https://issues.apache.org/jira/secure/projectavatar?size=small&pid=10594&avatarId=10122","16x16":"https://issues.apache.org/jira/secure/projectavatar?size=xsmall&pid=10594&avatarId=10122","32x32":"https://issues.apache.org/jira/secure/projectavatar?size=medium&pid=10594&avatarId=10122"},"projectCategory":{"self":"https://issues.apache.org/jira/rest/api/2/projectCategory/10090","id":"10090","description":"DB related projects","name":"DB"}},"fixVersions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12313111","id":"12313111","description":"","name":"10.4.1.3","archived":false,"released":true,"releaseDate":"2008-04-24"},{"self":"https://issues.apache.org/jira/rest/api/2/version/12313771","id":"12313771","description":"build: 764942, 14/Apr/09. Voted release: 28/Apr/09. Announced: 1/May/09","name":"10.5.1.1","archived":false,"released":true,"releaseDate":"2009-04-28"}],"aggregatetimespent":null,"resolution":{"self":"https://issues.apache.org/jira/rest/api/2/resolution/1","id":"1","description":"A fix for this issue is checked into the tree and tested.","name":"Fixed"},"customfield_12310220":"2008-03-31 12:41:42.882","customfield_12312322":null,"customfield_12312323":null,"customfield_12310420":"23726","customfield_12310222":"3_*:*_1_*:*_240826313_*|*_1_*:*_1_*:*_24130482_*|*_6_*:*_1_*:*_0_*|*_5_*:*_1_*:*_24559937030","customfield_12312320":null,"customfield_12312321":null,"customfield_12312120":null,"customfield_12312121":null,"resolutiondate":"2008-04-03T08:06:27.712+0000","workratio":-1,"customfield_12312328":null,"customfield_12312329":null,"customfield_12312326":null,"customfield_12310300":null,"customfield_12312327":null,"customfield_12312324":null,"customfield_12312325":null,"lastViewed":null,"watches":{"self":"https://issues.apache.org/jira/rest/api/2/issue/DERBY-3582/watchers","watchCount":0,"isWatching":false},"created":"2008-03-31T06:30:30.917+0000","priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/4","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/minor.png","name":"Minor","id":"4"},"labels":[],"customfield_12312333":null,"customfield_12312334":null,"customfield_12310310":"1.0","customfield_12312331":null,"customfield_12312332":null,"timeestimate":null,"aggregatetimeoriginalestimate":null,"versions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12313771","id":"12313771","description":"build: 764942, 14/Apr/09. Voted release: 28/Apr/09. Announced: 1/May/09","name":"10.5.1.1","archived":false,"released":true,"releaseDate":"2009-04-28"}],"customfield_12311120":null,"customfield_12312330":null,"issuelinks":[],"customfield_12312339":null,"assignee":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"customfield_12312337":null,"customfield_12312338":null,"updated":"2009-05-04T18:22:12.997+0000","customfield_12312335":null,"customfield_12311720":null,"customfield_12312336":null,"status":{"self":"https://issues.apache.org/jira/rest/api/2/status/6","description":"The issue is considered finished, the resolution is correct. Issues which are closed can be reopened.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Complete"}},"components":[{"self":"https://issues.apache.org/jira/rest/api/2/component/11415","id":"11415","name":"Services"}],"timeoriginalestimate":null,"description":"An IndexOutOfBoundsException was thrown during suites.All when testing a patch for DERBY-3576.\nI have only seen it once, and I was unable to reproduce. I'm logging the issue to track it and preserve the stack trace.\nSee the environment field for version information.\n\nI think only the first error is important, the rest (except the management test) is due to the first one.\nTime: 6,901.837\nThere were 5 errors:\n1) testFiringConstraintOrder(org.apache.derbyTesting.functionTests.tests.lang.TriggerTest)java.sql.SQLException: Java exception: 'Index: 0, Size: 0: java.lang.IndexOutOfBoundsException'.\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:95)\n\tat org.apache.derby.impl.jdbc.Util.newEmbedSQLException(Util.java:87)\n\tat org.apache.derby.impl.jdbc.Util.javaException(Util.java:244)\n\tat org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(TransactionResourceImpl.java:403)\n\tat org.apache.derby.impl.jdbc.TransactionResourceImpl.handleException(TransactionResourceImpl.java:346)\n\tat org.apache.derby.impl.jdbc.EmbedConnection.handleException(EmbedConnection.java:2083)\n\tat org.apache.derby.impl.jdbc.ConnectionChild.handleException(ConnectionChild.java:81)\n\tat org.apache.derby.impl.jdbc.EmbedDatabaseMetaData.getPreparedQueryUsingSystemTables(EmbedDatabaseMetaData.java:3498)\n\tat org.apache.derby.impl.jdbc.EmbedDatabaseMetaData.getPreparedQuery(EmbedDatabaseMetaData.java:3541)\n\tat org.apache.derby.impl.jdbc.EmbedDatabaseMetaData.getPreparedQuery(EmbedDatabaseMetaData.java:3566)\n\tat org.apache.derby.impl.jdbc.EmbedDatabaseMetaData.doGetProcs(EmbedDatabaseMetaData.java:1506)\n\tat org.apache.derby.impl.jdbc.EmbedDatabaseMetaData.getProcedures(EmbedDatabaseMetaData.java:1447)\n\tat org.apache.derbyTesting.junit.JDBC.dropSchema(JDBC.java:259)\n\tat org.apache.derbyTesting.functionTests.tests.lang.TriggerTest.tearDown(TriggerTest.java:103)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:103)\n\tat junit.extensions.TestDecorator.basicRun(TestDecorator.java:24)\n\tat junit.extensions.TestSetup$1.protect(TestSetup.java:21)\n\tat junit.extensions.TestSetup.run(TestSetup.java:25)\n\tat org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)\nCaused by: java.sql.SQLException: Java exception: 'Index: 0, Size: 0: java.lang.IndexOutOfBoundsException'.\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(SQLExceptionFactory.java:45)\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(SQLExceptionFactory40.java:119)\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:70)\n\t... 35 more\nCaused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0\n\tat java.util.ArrayList.RangeCheck(ArrayList.java:547)\n\tat java.util.ArrayList.get(ArrayList.java:322)\n\tat org.apache.derby.impl.services.cache.ClockPolicy.moveHand(ClockPolicy.java:364)\n\tat org.apache.derby.impl.services.cache.ClockPolicy.rotateClock(ClockPolicy.java:404)\n\tat org.apache.derby.impl.services.cache.ClockPolicy.insertEntry(ClockPolicy.java:176)\n\tat org.apache.derby.impl.services.cache.ConcurrentCache.insertIntoFreeSlot(ConcurrentCache.java:208)\n\tat org.apache.derby.impl.services.cache.ConcurrentCache.find(ConcurrentCache.java:284)\n\tat org.apache.derby.impl.sql.catalog.DataDictionaryImpl.getSPSDescriptor(DataDictionaryImpl.java:3529)\n\tat org.apache.derby.impl.jdbc.EmbedDatabaseMetaData.prepareSPS(EmbedDatabaseMetaData.java:3632)\n\tat org.apache.derby.impl.jdbc.EmbedDatabaseMetaData.getPreparedQueryUsingSystemTables(EmbedDatabaseMetaData.java:3493)\n\t... 28 more\n2) testClobInTriggerTable(org.apache.derbyTesting.functionTests.tests.lang.TriggerTest)java.sql.SQLException: PROCEDURE 'TRIGGER_LOG_INFO' already exists.\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:95)\n\tat org.apache.derby.impl.jdbc.Util.generateCsSQLException(Util.java:201)\n\tat org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(TransactionResourceImpl.java:391)\n\tat org.apache.derby.impl.jdbc.TransactionResourceImpl.handleException(TransactionResourceImpl.java:346)\n\tat org.apache.derby.impl.jdbc.EmbedConnection.handleException(EmbedConnection.java:2083)\n\tat org.apache.derby.impl.jdbc.ConnectionChild.handleException(ConnectionChild.java:81)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(EmbedStatement.java:1325)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.execute(EmbedStatement.java:625)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.executeUpdate(EmbedStatement.java:175)\n\tat org.apache.derbyTesting.functionTests.tests.lang.TriggerTest.setUp(TriggerTest.java:91)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:103)\n\tat junit.extensions.TestDecorator.basicRun(TestDecorator.java:24)\n\tat junit.extensions.TestSetup$1.protect(TestSetup.java:21)\n\tat junit.extensions.TestSetup.run(TestSetup.java:25)\n\tat org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)\nCaused by: java.sql.SQLException: PROCEDURE 'TRIGGER_LOG_INFO' already exists.\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(SQLExceptionFactory.java:45)\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(SQLExceptionFactory40.java:119)\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:70)\n\t... 31 more\nCaused by: ERROR X0Y68: PROCEDURE 'TRIGGER_LOG_INFO' already exists.\n\tat org.apache.derby.iapi.error.StandardException.newException(StandardException.java:303)\n\tat org.apache.derby.impl.sql.execute.CreateAliasConstantAction.executeConstantAction(CreateAliasConstantAction.java:237)\n\tat org.apache.derby.impl.sql.execute.MiscResultSet.open(MiscResultSet.java:64)\n\tat org.apache.derby.impl.sql.GenericPreparedStatement.execute(GenericPreparedStatement.java:372)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(EmbedStatement.java:1235)\n\t... 25 more\n3) testBlobInTriggerTable(org.apache.derbyTesting.functionTests.tests.lang.TriggerTest)java.sql.SQLException: PROCEDURE 'TRIGGER_LOG_INFO' already exists.\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:95)\n\tat org.apache.derby.impl.jdbc.Util.generateCsSQLException(Util.java:201)\n\tat org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(TransactionResourceImpl.java:391)\n\tat org.apache.derby.impl.jdbc.TransactionResourceImpl.handleException(TransactionResourceImpl.java:346)\n\tat org.apache.derby.impl.jdbc.EmbedConnection.handleException(EmbedConnection.java:2083)\n\tat org.apache.derby.impl.jdbc.ConnectionChild.handleException(ConnectionChild.java:81)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(EmbedStatement.java:1325)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.execute(EmbedStatement.java:625)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.executeUpdate(EmbedStatement.java:175)\n\tat org.apache.derbyTesting.functionTests.tests.lang.TriggerTest.setUp(TriggerTest.java:91)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:103)\n\tat junit.extensions.TestDecorator.basicRun(TestDecorator.java:24)\n\tat junit.extensions.TestSetup$1.protect(TestSetup.java:21)\n\tat junit.extensions.TestSetup.run(TestSetup.java:25)\n\tat org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)\nCaused by: java.sql.SQLException: PROCEDURE 'TRIGGER_LOG_INFO' already exists.\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(SQLExceptionFactory.java:45)\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(SQLExceptionFactory40.java:119)\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:70)\n\t... 31 more\nCaused by: ERROR X0Y68: PROCEDURE 'TRIGGER_LOG_INFO' already exists.\n\tat org.apache.derby.iapi.error.StandardException.newException(StandardException.java:303)\n\tat org.apache.derby.impl.sql.execute.CreateAliasConstantAction.executeConstantAction(CreateAliasConstantAction.java:237)\n\tat org.apache.derby.impl.sql.execute.MiscResultSet.open(MiscResultSet.java:64)\n\tat org.apache.derby.impl.sql.GenericPreparedStatement.execute(GenericPreparedStatement.java:372)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(EmbedStatement.java:1235)\n\t... 25 more\n4) testUpdateTriggerOnClobColumn(org.apache.derbyTesting.functionTests.tests.lang.TriggerTest)java.sql.SQLException: PROCEDURE 'TRIGGER_LOG_INFO' already exists.\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:95)\n\tat org.apache.derby.impl.jdbc.Util.generateCsSQLException(Util.java:201)\n\tat org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(TransactionResourceImpl.java:391)\n\tat org.apache.derby.impl.jdbc.TransactionResourceImpl.handleException(TransactionResourceImpl.java:346)\n\tat org.apache.derby.impl.jdbc.EmbedConnection.handleException(EmbedConnection.java:2083)\n\tat org.apache.derby.impl.jdbc.ConnectionChild.handleException(ConnectionChild.java:81)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(EmbedStatement.java:1325)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.execute(EmbedStatement.java:625)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.executeUpdate(EmbedStatement.java:175)\n\tat org.apache.derbyTesting.functionTests.tests.lang.TriggerTest.setUp(TriggerTest.java:91)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:103)\n\tat junit.extensions.TestDecorator.basicRun(TestDecorator.java:24)\n\tat junit.extensions.TestSetup$1.protect(TestSetup.java:21)\n\tat junit.extensions.TestSetup.run(TestSetup.java:25)\n\tat org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)\nCaused by: java.sql.SQLException: PROCEDURE 'TRIGGER_LOG_INFO' already exists.\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(SQLExceptionFactory.java:45)\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(SQLExceptionFactory40.java:119)\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:70)\n\t... 31 more\nCaused by: ERROR X0Y68: PROCEDURE 'TRIGGER_LOG_INFO' already exists.\n\tat org.apache.derby.iapi.error.StandardException.newException(StandardException.java:303)\n\tat org.apache.derby.impl.sql.execute.CreateAliasConstantAction.executeConstantAction(CreateAliasConstantAction.java:237)\n\tat org.apache.derby.impl.sql.execute.MiscResultSet.open(MiscResultSet.java:64)\n\tat org.apache.derby.impl.sql.GenericPreparedStatement.execute(GenericPreparedStatement.java:372)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(EmbedStatement.java:1235)\n\t... 25 more\n5) testTypesInActionStatement(org.apache.derbyTesting.functionTests.tests.lang.TriggerTest)java.sql.SQLException: PROCEDURE 'TRIGGER_LOG_INFO' already exists.\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:95)\n\tat org.apache.derby.impl.jdbc.Util.generateCsSQLException(Util.java:201)\n\tat org.apache.derby.impl.jdbc.TransactionResourceImpl.wrapInSQLException(TransactionResourceImpl.java:391)\n\tat org.apache.derby.impl.jdbc.TransactionResourceImpl.handleException(TransactionResourceImpl.java:346)\n\tat org.apache.derby.impl.jdbc.EmbedConnection.handleException(EmbedConnection.java:2083)\n\tat org.apache.derby.impl.jdbc.ConnectionChild.handleException(ConnectionChild.java:81)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(EmbedStatement.java:1325)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.execute(EmbedStatement.java:625)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.executeUpdate(EmbedStatement.java:175)\n\tat org.apache.derbyTesting.functionTests.tests.lang.TriggerTest.setUp(TriggerTest.java:91)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:103)\n\tat junit.extensions.TestDecorator.basicRun(TestDecorator.java:24)\n\tat junit.extensions.TestSetup$1.protect(TestSetup.java:21)\n\tat junit.extensions.TestSetup.run(TestSetup.java:25)\n\tat org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)\nCaused by: java.sql.SQLException: PROCEDURE 'TRIGGER_LOG_INFO' already exists.\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory.getSQLException(SQLExceptionFactory.java:45)\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.wrapArgsForTransportAcrossDRDA(SQLExceptionFactory40.java:119)\n\tat org.apache.derby.impl.jdbc.SQLExceptionFactory40.getSQLException(SQLExceptionFactory40.java:70)\n\t... 31 more\nCaused by: ERROR X0Y68: PROCEDURE 'TRIGGER_LOG_INFO' already exists.\n\tat org.apache.derby.iapi.error.StandardException.newException(StandardException.java:303)\n\tat org.apache.derby.impl.sql.execute.CreateAliasConstantAction.executeConstantAction(CreateAliasConstantAction.java:237)\n\tat org.apache.derby.impl.sql.execute.MiscResultSet.open(MiscResultSet.java:64)\n\tat org.apache.derby.impl.sql.GenericPreparedStatement.execute(GenericPreparedStatement.java:372)\n\tat org.apache.derby.impl.jdbc.EmbedStatement.executeStatement(EmbedStatement.java:1235)\n\t... 25 more\nThere was 1 failure:\n1) testStartStopManagementFromApplication(org.apache.derbyTesting.functionTests.tests.management.ManagementMBeanTest)junit.framework.AssertionFailedError: expected:<2> but was:<5>\n\tat org.apache.derbyTesting.functionTests.tests.management.ManagementMBeanTest.startStopManagement(ManagementMBeanTest.java:86)\n\tat org.apache.derbyTesting.functionTests.tests.management.ManagementMBeanTest.testStartStopManagementFromApplication(ManagementMBeanTest.java:56)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n\tat sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n\tat sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n\tat org.apache.derbyTesting.junit.BaseTestCase.runBare(BaseTestCase.java:103)\n\tat junit.extensions.TestDecorator.basicRun(TestDecorator.java:24)\n\tat junit.extensions.TestSetup$1.protect(TestSetup.java:21)\n\tat junit.extensions.TestSetup.run(TestSetup.java:25)\n\tat org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)\n\tat junit.extensions.TestDecorator.basicRun(TestDecorator.java:24)\n\tat junit.extensions.TestSetup$1.protect(TestSetup.java:21)\n\tat junit.extensions.TestSetup.run(TestSetup.java:25)\n\tat junit.extensions.TestDecorator.basicRun(TestDecorator.java:24)\n\tat junit.extensions.TestSetup$1.protect(TestSetup.java:21)\n\tat junit.extensions.TestSetup.run(TestSetup.java:25)\n\tat org.apache.derbyTesting.junit.BaseTestSetup.run(BaseTestSetup.java:57)\n\nFAILURES!!!\nTests run: 9674,  Failures: 1,  Errors: 5","customfield_10010":null,"timetracking":{},"customfield_12312026":null,"customfield_12310200":null,"customfield_12312023":null,"customfield_12312024":null,"customfield_12312340":null,"aggregatetimeestimate":null,"customfield_12312022":null,"customfield_12312341":null,"customfield_12312220":null,"customfield_12310921":null,"customfield_12310920":"39304","summary":"IndexOutOfBoundsError in ClockPolicy.moveHand","creator":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kristwaa","name":"kristwaa","emailAddress":"kristwaa at apache dot org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=kristwaa&avatarId=14736","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=kristwaa&avatarId=14736","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=kristwaa&avatarId=14736","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=kristwaa&avatarId=14736"},"displayName":"Kristian Waagan","active":true},"subtasks":[],"customfield_12310090":null,"reporter":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kristwaa","name":"kristwaa","emailAddress":"kristwaa at apache dot org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=kristwaa&avatarId=14736","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=kristwaa&avatarId=14736","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=kristwaa&avatarId=14736","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=kristwaa&avatarId=14736"},"displayName":"Kristian Waagan","active":true},"customfield_12310291":null,"customfield_12310290":null,"aggregateprogress":{"progress":0,"total":0},"environment":"Dual CPU 2.4 GHz Opteron (x86)\n\nSun Solaris 10u2\n\njava version \"1.6.0_04-p\"\nJava(TM) SE Runtime Environment (build 1.6.0_04-p-b03)\nJava HotSpot(TM) Client VM (build 11.0-b09, mixed mode)\n\nDerby trunk version: 10.5.0.0 alpha - (642119M)","customfield_12311020":null,"customfield_12310050":null,"duedate":null,"customfield_12310250":null,"progress":{"progress":0,"total":0},"comment":{"startAt":0,"maxResults":6,"total":6,"comments":[{"self":"https://issues.apache.org/jira/rest/api/2/issue/12392702/comment/12583664","id":"12583664","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"I think I see how this could happen. ClockPolicy.rotateClock() is never called when the ClockPolicy.clock is empty, therefore ClockPolicy.moveHand() doesn't check that the clock actually contains entries before calling ArrayList.get(). However, in this case the clock was empty in moveHand(), which means that it must have shrunk while rotateClock() was executing. The only code that can shrink the clock so much that it becomes empty is ClockPolicy.trimMe(), which was removed from trunk in revision 642755. When I back out that commit and run TriggerTest, I see in some of the runs that trimMe() is invoked and makes the clock in the SPS descriptor cache empty. I'm not able to reproduce the IndexOutOfBoundsException, though. I think another thread needs to be involved to make that happen.\r\n\r\nRegardless of whether or not the problem was fixed in revision 642755, it's probably a good idea to make moveHand() and rotateClock() check if the clock is empty and deal with the situation if it occurs.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2008-03-31T12:41:42.882+0000","updated":"2008-03-31T12:41:42.882+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12392702/comment/12583690","id":"12583690","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"Attached is a patch that makes ClockPolicy.moveHand() and ClockPolicy.rotateClock() handle that the clock becomes empty while they are executing. All that is needed is that they return null, and then ClockPolicy.insertEntry() will understand that there are no entries that can be reused, and it will add a new one. It is probably not needed after revision 642755, but I think it is safer to do the check anyway to minimize the risk of introducing a similar bug by accident later.\r\n\r\nI have started the full regression test suite.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2008-03-31T13:59:13.736+0000","updated":"2008-03-31T13:59:13.736+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12392702/comment/12583816","id":"12583816","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"Derbyall and suites.All passed (except ManagementMBeanTest, DERBY-3561).","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2008-03-31T20:09:33.612+0000","updated":"2008-03-31T20:09:33.612+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12392702/comment/12584535","id":"12584535","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"Committed revision 643870.\r\nWill also back-port to 10.4.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2008-04-02T12:13:41.671+0000","updated":"2008-04-02T12:13:41.671+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12392702/comment/12585004","id":"12585004","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"body":"Committed to 10.4 with revision 644211.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=knutanders","name":"knutanders","emailAddress":"knut dot hatlen at oracle dot com","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Knut Anders Hatlen","active":true},"created":"2008-04-03T08:06:27.625+0000","updated":"2008-04-03T08:06:27.625+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12392702/comment/12662963","id":"12662963","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kristwaa","name":"kristwaa","emailAddress":"kristwaa at apache dot org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=kristwaa&avatarId=14736","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=kristwaa&avatarId=14736","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=kristwaa&avatarId=14736","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=kristwaa&avatarId=14736"},"displayName":"Kristian Waagan","active":true},"body":"Closing the issue.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=kristwaa","name":"kristwaa","emailAddress":"kristwaa at apache dot org","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=kristwaa&avatarId=14736","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=kristwaa&avatarId=14736","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=kristwaa&avatarId=14736","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=kristwaa&avatarId=14736"},"displayName":"Kristian Waagan","active":true},"created":"2009-01-12T14:18:44.732+0000","updated":"2009-01-12T14:18:44.732+0000"}]},"votes":{"self":"https://issues.apache.org/jira/rest/api/2/issue/DERBY-3582/votes","votes":0,"hasVoted":false},"worklog":{"startAt":0,"maxResults":20,"total":0,"worklogs":[]},"customfield_12311820":"0|i073bj:"}}